### PR TITLE
ref(tagstore): Add `TagStorage.get_event_tag_qs`

### PR DESCRIPTION
**Pull request by tkaemming**

### CODE REVIEW

File: src/sentry/tagstore/legacy/backend.py [hunk 1]
```diff
@@ -628,6 +628,9 @@ def get_group_tag_value_qs(self, project_id, group_id, environment_id, key, valu
 
         return queryset
 
+    def get_event_tag_qs(self, project_id, environment_id, key, value):
+        raise NotImplementedError  # there is no index that can appopriate satisfy this query
```
Comment (tkaemming) [inline src/sentry/tagstore/legacy/backend.py]:
```
Indexes:
    "sentry_eventtag_pkey" PRIMARY KEY, btree (id)
    "sentry_eventtag_event_id_430cef8ef4186908_uniq" UNIQUE CONSTRAINT, btree (event_id, key_id, value_id)
    "sentry_eventtag_date_added" btree (date_added)
    "sentry_eventtag_group_id_5ad9abfe8e1fa62b" btree (group_id, key_id, value_id)```

File: src/sentry/tagstore/v2/backend.py [hunk 2]
```diff
@@ -815,6 +815,17 @@ def get_group_tag_value_qs(self, project_id, group_id, environment_id, key, valu
...
+    def get_event_tag_qs(self, project_id, environment_id, key, value):
+        qs = EventTag.objects.filter(
+            project_id=project_id,
+            key__key=key,
+            value__value=value,
+        )
+
+        qs = self._add_environment_filter(qs, environment_id)
+
+        return qs
```
Comment (tkaemming) [inline src/sentry/tagstore/v2/backend.py]: This should cause this index to be used:
```
"tagstore_eventtag_project_id_3479ea564384cd3f" btree (project_id, key_id, value_id)
```

**Review by tkaemming — COMMENTED** [review]
This is supporting work for adding `date_{from,to}` support to GH-7565.

I don't like exposing the model at this level, but it seemed worse to allow passing in arbitrary filter constraints and grouping keys (to facilitate the event time range search.)

File: src/sentry/tagstore/legacy/backend.py (hunk 1 above)
Comment (tkaemming) [reply src/sentry/tagstore/legacy/backend.py]: This also implies that environment-based `date_{from,to}`/`event.timestamp` queries won't work with the legacy backend, if that wasn't clearly articulated.

Comment (bretthoerner) [reply src/sentry/tagstore/legacy/backend.py]: I guess it's part of a future PR, but what does this mean for... calling the method? Will we only deploy the future code (that calls this) when v2 is in use? Or will we detect the backend and do something else, or?

**Review by bretthoerner — APPROVED** [review]
Had a question but it seems fine to me so far.

I'm working on removing EventTag.key too.

File: src/sentry/tagstore/legacy/backend.py (hunk 1 above)
Comment (tkaemming) [reply src/sentry/tagstore/legacy/backend.py]: This function should only be hit if:

- The legacy tagstore backend is used (obviously),
- The search backend changes (GH-7565) have landed (this should happen soon),
- The `environment` parameter is sent to the search backend (this depends on UI feature flag.)

I guess there's not really a great way to enforce that the UI flag isn't switched over before the backend is switched, but I don't anticipate us enabling the feature before the tagstore backend has been fully switched over. For on-premise, though…

**Pull request merged**